Handling of sequential memory modifications.

abdelhamidbakhta opened the pull request:
<!-- Thanks for sending a pull request! Please check out our contribution guidelines: -->
<!-- https://github.com/hyperledger/besu/blob/master/CONTRIBUTING.md -->

## PR description
Tracing API should properly handle  sequential memory modifications.

Added one smart contract in genesis:
[1] PUSH1 0x01
[3] PUSH1 0x02
[4] MSTORE8
[6] PUSH1 0x03
[8] PUSH1 0x04
[9] MSTORE8
[11] PUSH1 0x01
[13] PUSH1 0x05
[14] MSTORE8
## Fixed Issue(s)
<!-- Please link to fixed issue(s) here using format: fixes #<issue number> -->
<!-- Example: "fixes #2" -->
fixes https://pegasys1.atlassian.net/browse/PIE-2225
fixes https://pegasys1.atlassian.net/browse/PIE-2257

Signed-off-by: Abdelhamid Bakhta <abdelhamid.bakhta@consensys.net>

shemnon commented on ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/vm/VmTraceGenerator.java:
```diff
@@ -101,6 +98,9 @@ private boolean mustIgnore(final TraceFrame frame) {
     if ("STOP".equals(frame.getOpcode()) && transactionTrace.getTraceFrames().size() == 1) {
```
Style... could these be if (...) { return ... } else if (...) { return ... } else { return ...} - having the returns at the same indent and in an else-if chain I find increases readability.  It makes the effect explicit, if X then A, if Y then B, otherwise C.

abdelhamidbakhta replied:
Sure. Fixed.

abdelhamidbakhta commented:
> Since we updated the blocks and genesis.json I was expecting there would also be a corresponding `.bin` file.
> 
> Before you commit can you update the `.bin` file or validate that it is a no-diff?

Nice catch. I added the relevant .bin
